src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InitializeHandler.cs

```diff
@@ -313,6 +319,12 @@ private bool GetMergedRenameProvider()
             return _serverCapabilities.Any(s => s.Capabilities.RenameProvider?.Value is bool isRenameSupported && isRenameSupported);
         }
 
+        private VSInternalInlineCompletionOptions GetMergedInlineCompletionProvider()
+        {
+            var regexes = _serverCapabilities.Where(s => s.Capabilities.InlineCompletionOptions != null).Select(s => s.Capabilities.InlineCompletionOptions.Pattern.ToString());
+            return new VSInternalInlineCompletionOptions { Pattern = new Regex(string.Join("|", regexes)) };
```

Comment:
Nit: but on the off chance that one of the Regexes ends up null or empty we might specify that the `string.Join` should drop empty entries.